File: PHP/ValidateStackSequencesTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit ValidateStackSequencesTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("ValidateStackSequences.php");

class ValidateStackSequencesTest extends TestCase {

    public function testValidateStackSequences() {
        $obj = new ValidateStackSequences();
        //Test 1
        $want = true;
        $got = $obj->validateStackSequences( [1,2,3,4,5], [4,5,3,2,1] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = false;
        $got = $obj->validateStackSequences( [1,2,3,4,5], [4,3,5,1,2] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/ValidateStackSequences_test.go
/**
 * @version GO 1.20.3
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestValidateStackSequences
 */
package GO

import (
	"reflect"
	"testing"
)

func TestValidateStackSequences(t *testing.T) {
	tables := []struct {
		want   bool
		pushed []int
		popped []int
	}{
		{true, []int{1, 2, 3, 4, 5}, []int{4, 5, 3, 2, 1}},
		{false, []int{1, 2, 3, 4, 5}, []int{4, 3, 5, 1, 2}},
	}

	for _, table := range tables {
		got := validateStackSequences(table.pushed, table.popped)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %t but the return was this: %t to %d", table.want, got, table.pushed)
		}
	}
}
